crates/containerd-shim-wasm/src/sys/unix/container/executor.rs

```diff
@@ -62,17 +70,22 @@ impl<E: Engine> LibcontainerExecutor for Executor<E> {
 }
 
 impl<E: Engine> Executor<E> {
-    pub fn new(engine: E, stdio: Stdio) -> Self {
+    pub fn new(engine: E, stdio: Stdio, oci_artifacts: Option<Vec<OciArtifact>>) -> Self {
         Self {
             engine,
             stdio,
             inner: Default::default(),
+            oci_artifacts,
         }
     }
 
     fn inner(&self, spec: &Spec) -> &InnerExecutor {
+        let is_wasm = self.oci_artifacts.is_some();
         self.inner.get_or_init(|| {
-            if is_linux_container(spec).is_ok() {
+            // if the spec has oci annotations we know it is wasm so short circuit checks
```

Comment:
**question:** I'm not sure what the right thing to do is here.
If we have wasm annotations, we don't need to check for linux containers, I agree.
But a runtime may want to do other checks before deciding it can handle it? e.g., in the future they may want to check if a component is a _reactor_ or a _command_?

Reply:
good point, I was trying to avoid the extra cycles of having to parse the executable files, but maybe it makes more sense to have this in the default `can_handle` of `trait Engine` so it can be overridden if required?

Reply:
I am happy to keep it as is.
We might want to revisit this in the future.